src/libs/comms/serialization/ClusterFunSerializer.ts

```diff
@@ -179,49 +178,21 @@ export default class ClusterFunSerializer {
     }
 
     private deserializePartsInternal(input: string, includePayload: boolean): RawMessagePacket<unknown>{
-        // Use @streamparser/json to parse our concatenated JSON message.
+        const splitLocation = input.indexOf("}^{") + 1;
```

Comment:
Two issues:
1. The `serializeParts` function should include this `^` character between the header and payload
2. The search itself should be `input.substring(0, MAX_HEADER_LENGTH).indexOf("}^{") + 1`. While not as bad as when we were doing JSON.parse() on the whole string, forcing the server to search through the whole string for a header split will still cause large payloads to slow it down.

Reply:
Also, if we eventually want to support payloads that are not JSON, our search pattern should not depend on an opening bracket to begin the payload. (We have one test that makes this assumption).